### PR TITLE
UIOR-831 UIOR-832 Allow user to add Package titles

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UIOR-831
https://issues.folio.org/browse/UIOR-832

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by usavkov-epam — APPROVED** [review]

File: src/components/POLine/LineLinkedInstances/LineLinkedInstances.js [hunk 1]
```diff
@@ -35,18 +46,64 @@ const resultFormatter = {
...
+    const title = getTitle(instance, line.id);
+
+    return mutateTitle(title)
+      .then(() => {
+        showCallout({
+          messageId: 'ui-orders.title.actions.save.success',
+          values: { title: title.title, poLineNumber: line.poLineNumber },
+        });
+        refetch();
+      }, (async ({ response }) => {
```
Comment (NikitaSedyx) [inline src/components/POLine/LineLinkedInstances/LineLinkedInstances.js:69]: catch?

File: src/components/POLine/LineLinkedInstances/LineLinkedInstances.js [hunk 2]
```diff
@@ -35,18 +46,64 @@ const resultFormatter = {
...
-  const { isLoading, linkedInstances } = useLinkedInstances(line);
+  const showCallout = useShowCallout();
+  const { isLoading, linkedInstances, refetch } = useLinkedInstances(line);
+  const { mutateTitle } = useTitleMutation();
 
   useEffect(() => {
     toggleSection({ id: ACCORDION_ID.linkedInstances, isOpened: Boolean(isLoading || linkedInstances?.length) });
   }, [toggleSection, isLoading, linkedInstances]);
 
+  const onAddInstance = useCallback((instance) => {
```
Comment (NikitaSedyx) [inline src/components/POLine/LineLinkedInstances/LineLinkedInstances.js:59]: I think you can omit useCallback here

File: src/components/POLine/POLineView.js [hunk 3]
```diff
@@ -429,10 +436,13 @@ const POLineView = ({
             lineId={line.id}
           />
 
-          <LineLinkedInstances
-            line={line}
-            toggleSection={onToggleSection}
-          />
+          {!line.isPackage && (
```
Comment (NikitaSedyx) [inline src/components/POLine/POLineView.js:439]: so was it agreed to hide for non-pacakge?

File: src/components/POLine/LineLinkedInstances/utils.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+const ALLOWED_YEAR_LENGTH = 4;
+
+export const getTitle = (instance, lineId) => {
```
Comment (NikitaSedyx) [inline src/components/POLine/LineLinkedInstances/utils.js:3]: get is incorrect name here

File: src/components/POLine/POLineView.js (hunk 3 above)
Comment (AndreiShumski2) [reply src/components/POLine/POLineView.js:439]: We agreed to display Package titles accordion on package POL under Item details accordion and Linked instance accordion for regular POL on the previous position

**Pull request merged**